src/ucp/wireup/select.c

```diff
@@ -1193,14 +1309,18 @@ static ucs_status_t ucp_wireup_add_tag_lane(ucp_ep_h ep, unsigned address_count,
 
     /* Do not add tag offload lane, if selected tag lane score is lower
      * than AM score. In this case AM will be used for tag macthing. */
-    status = ucp_wireup_select_transport(ep, address_list, address_count,
-                                         &criteria, -1, -1, -1, -1, 0,
-                                         &select_info);
+    status = ucp_wireup_select_lane(ep, address_count, address_list,
+                                    &criteria, -1, -1, -1, -1, 0,
+                                    &select_info);
     /* Break if: */
     /* - transport selection wasn't OK */
     if ((status != UCS_OK) ||
         /* - the TAG transport is worse than the AM transport */
-        (ucp_wireup_score_cmp(select_info.score, am_info->score <= 0))) {
+        (ucp_wireup_score_cmp(select_info.score, am_info->score <= 0)) ||
+        /* - the TAG transport isn't scalable enough, but the
+         *   selected AM transport provides better scalability */
+        (!ucp_wireup_check_scalability(context, select_info.max_num_eps,
```

Comment:
let's select AM instead of tag only if it matches scalability criteria (not just better than TAG lane)

Reply:
do you mean
```
!ucp_is_scalable_transport(context, select_info.max_num_eps) &&
ucp_is_scalable_transport(context, am_info->max_num_eps)
``` ?

I think it's better to have the same behavior for all lanes. Let's do it for all lanes? i.e. update `ucp_wireup_check_scalability`

Reply:
ok